### PR TITLE
Provide context for errors when reading status line and headers

### LINKED ISSUE #320: "Network Error: stream did not contain valid UTF-8" error, even with 'charset' enabled
On some websites, e.g. paneliste.com, ureq fails with the following error:

> Network Error: stream did not contain valid UTF-8

despite the `charset` feature being enabled.

Firefox and curl work fine.

There's 96 such websites in the top million (I'm using [Tranco list generated on the 3rd of February](https://tranco-list.eu/list/3G6L)).

Archive with all occurrences: [ureq-did-not-contain-valid-utf8.tar.gz](https://github.com/algesten/ureq/files/5983425/ureq-did-not-contain-valid-utf8.tar.gz)

Code used for testing: https://github.com/Shnatsel/rust-http-clients-smoke-test/blob/f206362f2e81521bbefb84007cdd25242f6db590/ureq-smoke-test/src/main.rs

### PR DESCRIPTION
**Pull request by algesten**
The HTTP spec allows for non-ascii values both in the status line and
in headers. ureq does not handle that, we can however provide better
context for when it happens.

Close #320

### CODE REVIEW

File: src/response.rs [hunk 1]
```diff
@@ -539,9 +539,30 @@ impl FromStr for Response {
...
+    if let Err(e) = result {
+        // Provide context to errors encountered while reading the line.
+        // ureq does not currently handle non-ascii status lines and
+        // header values. For historical reasons, the HTTP spec does
+        // allow for characters in the range 0x80-0xff, but these are
+        // very rarely encountered in the wild.
+        // See https://github.com/algesten/ureq/issues/320
+        let reason = if e.kind() == io::ErrorKind::InvalidData {
+            format!(
+                "Invalid data in {} (see issue #320): {}",
```
Comment (jsha) [inline src/response.rs:555]: People are likely to see this error in contexts where it's not clear it comes from ureq, so `issue #320` will be confusing. I would recommend omitting it; if you'd like to include it, let's use a full URL.

File: src/response.rs [hunk 2]
```diff
@@ -539,9 +539,30 @@ impl FromStr for Response {
...
+                context,
+                e.to_string()
```
Comment (jsha) [inline src/response.rs:557]: I think if we add `.src(e)` to the error, we don't need to interpolate it as a string here.

**Review by jsha — CHANGES_REQUESTED** [review]
Looks good!  A couple of tweaks below.

File: src/response.rs (hunk 1 above)
Comment (algesten) [reply src/response.rs:555]: Ok!

File: src/response.rs (hunk 2 above)
Comment (algesten) [reply src/response.rs:557]: Ok!

**Comment by algesten** [comment]
@jsha This should be ok now. If you agree I squash, rebase and merge.

**Review by jsha — APPROVED** [review]

**Pull request merged**